homeassistant/components/sensor/coinmarketcap.py

```diff
@@ -73,7 +80,9 @@ def name(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return round(float(self._ticker.get('price_usd')), 2)
+        return round(float(self._ticker.get(
+            'price_' +
+            self._unit_of_measurement.lower())), 2)
```

Comment:
Use `.lower()` in init to only do it once. To keep it backwards compatible and not mess with the user's history, make sure that `return self._unit_of_measurement` stays upper case.